packages/ckeditor5-engine/tests/model/history.js

```diff
@@ -78,31 +82,55 @@ describe( 'History', () => {
 	} );
 
 	describe( 'getOperations', () => {
-		let ops;
+		it( 'should return only operations from given base version', () => {
+			history.addOperation( ( new Operation( 0 ) ) );
+			history.addOperation( ( new Operation( 1 ) ) );
+			history.addOperation( ( new Operation( 2 ) ) );
 
-		beforeEach( () => {
-			ops = getOperations();
+			const ops = history.getOperations( 1 );
 
-			for ( const op of ops ) {
-				history.addOperation( op );
-			}
+			expect( ops.length ).to.equal( 2 );
+			expect( ops[ 0 ].baseVersion ).to.equal( 1 );
+			expect( ops[ 1 ].baseVersion ).to.equal( 2 );
 		} );
 
-		it( 'should return only operations from given base version', () => {
-			const historyOperations = Array.from( history.getOperations( 1 ) );
+		it( 'should return only operations up to given base version', () => {
+			history.addOperation( ( new Operation( 0 ) ) );
+			history.addOperation( ( new Operation( 1 ) ) );
+			history.addOperation( ( new Operation( 2 ) ) );
+
+			const ops = history.getOperations( 1, 2 );
+
+			expect( ops.length ).to.equal( 1 );
+			expect( ops[ 0 ].baseVersion ).to.equal( 1 );
+		} );
 
-			expect( historyOperations ).to.deep.equal( ops.slice( 1 ) );
+		it( 'should return empty array if no operations match', () => {
+			history.addOperation( ( new Operation( 0 ) ) );
+			history.addOperation( ( new Operation( 1 ) ) );
+
+			expect( history.getOperations( 20 ).length ).to.equal( 0 );
+			expect( history.getOperations( -3, 0 ).length ).to.equal( 0 );
 		} );
 
-		it( 'should return only operations up to given base version', () => {
-			const historyOperations = Array.from( history.getOperations( 1, 2 ) );
+		it( 'should return correct values if history holds operations with negative base version', () => {
+			history.addOperation( ( new Operation( -2 ) ) );
+			history.addOperation( ( new Operation( -1 ) ) );
+			history.addOperation( ( new Operation( 0 ) ) );
+			history.addOperation( ( new Operation( 1 ) ) );
+			history.addOperation( ( new Operation( 2 ) ) );
 
-			expect( historyOperations ).to.deep.equal( ops.slice( 1, 2 ) );
+			expect( history.getOperations( -1, 2 ).length ).to.equal( 3 ); // -1, 0, 1.
```

Comment:
Maybe we could check returned operations whether they are correct instead of code comment?